flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/TemporalJoinITCase.scala

```diff
@@ -19,50 +19,100 @@
+    changelogRow("+I", "Yen", "no1", 1L),
+    changelogRow("+I", "RMB", "no1", 702L),
+    changelogRow("-U", "RMB", "no1", 702L),
+    changelogRow("+U", "RMB", "no1", 802L),
+    changelogRow("+I", "Euro", "no1", 118L),
+    changelogRow("+I", "US Dollar", "no2", 106L))
+
+  // test data for Event-Time temporal table join
+  val rowTimeOrderData = List(
+    changelogRow("+I", 1L, "Euro", "no1", 12L, "2020-08-15T00:01:00"),
+    changelogRow("+I", 2L, "US Dollar", "no1", 1L, "2020-08-15T00:02:00"),
+    changelogRow("+I", 3L, "RMB", "no1", 40L, "2020-08-15T00:03:00"),
+    changelogRow("+I", 4L, "Euro", "no1", 14L, "2020-08-16T00:02:00"),
+    changelogRow("-U", 2L, "US Dollar", "no1", 1L, "2020-08-16T00:03:00"),
+    changelogRow("+U", 2L, "US Dollar", "no1", 18L, "2020-08-16T00:03:00"),
+    changelogRow("+I", 5L, "RMB", "no1", 40L, "2020-08-16T00:03:00"),
+    changelogRow("+I", 6L, "RMB", "no1", 40L, "2020-08-16T00:04:00"),
+    changelogRow("-D", 6L, "RMB", "no1", 40L, "2020-08-16T00:04:00"))
+
+  val rowTimeCurrencyDataUsingMetaTime = List(
+    changelogRow("+I", "Euro", "no1", 114L, "2020-08-15T00:00:00"),
+    changelogRow("+I", "US Dollar", "no1", 102L, "2020-08-15T00:00:00"),
+    changelogRow("+I", "Yen", "no1", 1L, "2020-08-15T00:00:00"),
+    changelogRow("+I", "RMB", "no1", 702L, "2020-08-15T00:00:00"),
+    changelogRow("-U", "Euro", "no1", 114L, "2020-08-16T00:01:00"),
+    changelogRow("+U", "Euro",  "no1", 118L, "2020-08-16T00:01:00"),
+    changelogRow("-U", "US Dollar", "no1", 102L, "2020-08-16T00:02:00"),
+    changelogRow("+U", "US Dollar",  "no1", 106L, "2020-08-16T00:02:00"),
+    changelogRow("-D", "RMB", "no1", 708L, "2020-08-16T00:02:00"))
+
+  val rowTimeCurrencyDataUsingUpdateBeforeTime = List(
+    changelogRow("+I", "Euro", "no1", 114L, "2020-08-15T00:00:00"),
+    changelogRow("+I", "US Dollar", "no1", 102L, "2020-08-15T00:00:00"),
+    changelogRow("+I", "Yen", "no1", 1L, "2020-08-15T00:00:00"),
+    changelogRow("+I", "RMB", "no1", 702L, "2020-08-15T00:00:00"),
+    changelogRow("+I", "RMB", "no1", 708L, "2020-08-16T00:00:00"),
+    changelogRow("-U", "Euro", "no1", 114L, "2020-08-16T00:01:00"),
+    changelogRow("+U", "Euro",  "no1", 118L, "2020-08-16T00:01:00"),
+    changelogRow("-U", "US Dollar", "no1", 102L, "2020-08-15T00:00:00"),
+    changelogRow("+U", "US Dollar",  "no1", 106L, "2020-08-16T00:02:00"),
+    changelogRow("-D", "RMB", "no1", 702L, "2020-08-15T00:00:00"))
+
+  val upsertSourceCurrencyData = List(
+    changelogRow("+U", "Euro", "no1", 114L, "2020-08-15T00:00:00"),
+    changelogRow("+U", "US Dollar", "no1", 102L, "2020-08-15T00:00:00"),
+    changelogRow("+U", "Yen", "no1", 1L, "2020-08-15T00:00:00"),
+    changelogRow("+U", "RMB", "no1", 702L, "2020-08-15T00:00:00"),
+    changelogRow("+U", "US Dollar", "no1", 104L, "2020-08-16T00:00:00"),
+    changelogRow("-D", "RMB", "no1", 702L, "2020-08-15T00:00:00"),
+    changelogRow("+U", "RMB", "no1", 712L, "2020-08-16T00:00:00"))
```

Comment:
Would be better to have the same history data, then the result of temporal join would be more predicable.